js/packages/web/src/components/AuctionCard/index.tsx

```diff
@@ -516,74 +520,132 @@ export const AuctionCard = ({
+                    !auctionView.myBidderPot &&
+                    isAuctionManagerAuthorityNotWalletOwner;
+                  const allowBidToAuctionOwner =
+                    myPayingAccount &&
+                    !isAuctionManagerAuthorityNotWalletOwner &&
+                    isAuctionItemMaster;
+
+                  // Placing a "bid" of the full amount results in a purchase to redeem.
+                  if (
+                    instantSalePrice &&
+                    (allowBidToPublic || allowBidToAuctionOwner)
+                  ) {
+                    let bidTxid: string | undefined;
+
+                    try {
+                      const { amount, txid } = await sendPlaceBid(
+                        connection,
+                        wallet,
+                        myPayingAccount.pubkey,
+                        auctionView,
+                        accountByMint,
+                        instantSalePrice,
+                        'confirmed',
+                      );
+                      setLastBid({ amount });
+                      bidTxid = txid;
+                    } catch (e) {
+                      console.error('sendPlaceBid', e);
+                      return;
+                    }
+
+                    try {
+                      // Attempt to load the transaction, retrying up to 5 times
+                      retry: do {
+                        for (let i = 0; i < 5; ++i) {
+                          const tx = await connection.getTransaction(bidTxid, {
+                            commitment: 'confirmed',
+                          });
+
+                          const keys = tx?.transaction.message.accountKeys;
+
+                          if (!keys) {
+                            await new Promise(o => setTimeout(o, 2000));
+                            continue;
+                          }
+
+                          const patch = await loadMultipleAccounts(
+                            connection,
+                            keys.map(k => k.toBase58()),
+                            'confirmed',
+                          );
+
+                          const newState = patchState(patch);
+                          setShowBidModal(true);
+                          setLoading(true);
+
+                          {
+                            const auctionKey = auctionView.auction.pubkey;
+                            const auctionBidderKey = `${auctionKey}-${wallet.publicKey}`;
+
+                            auctionView.auction = newState.auctions[auctionKey];
+                            auctionView.myBidderPot =
+                              newState.bidderPotsByAuctionAndBidder[
+                                auctionBidderKey
+                              ];
+                            auctionView.myBidderMetadata =
+                              newState.bidderMetadataByAuctionAndBidder[
+                                auctionBidderKey
+                              ];
```

Comment:
If all these are available in the patch state on LN581 then we can remove the return of the new state and just use the patch state to adjust the auction view.